aspnetcore/fundamentals/servers/kestrel/endpoints.md

```diff
@@ -325,130 +450,129 @@ HTTPS options that can be overridden by SNI:
+#### SNI with `ServerOptionsSelectionCallback`
 
-The default value, `SslProtocols.None`, causes Kestrel to use the operating system defaults to choose the best protocol. Unless you have a specific reason to select a protocol, use the default.
+Kestrel supports additional dynamic TLS configuration via the `ServerOptionsSelectionCallback` callback. The callback is invoked once per connection to allow the app to inspect the host name and select the appropriate certificate and TLS configuration. Default certificates and `ConfigureHttpsDefaults` aren't used with this callback.
 
-## Client Certificates
+:::code language="csharp" source="~/fundamentals/servers/kestrel/samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_ServerOptionsSelectionCallback":::
 
-`ClientCertificateMode` configures the [client certificate requirements](xref:Microsoft.AspNetCore.Server.Kestrel.Https.ClientCertificateMode).
+#### SNI with `TlsHandshakeCallbackOptions`
 
-:::code language="csharp" source="~/fundamentals/servers/kestrel/samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_ConfigureHttpsDefaultsClientCertificateMode":::
+Kestrel supports additional dynamic TLS configuration via the `TlsHandshakeCallbackOptions.OnConnection` callback. The callback is invoked once per connection to allow the app to inspect the host name and select the appropriate certificate, TLS configuration, and other server options. Default certificates and `ConfigureHttpsDefaults` aren't used with this callback.
 
-```json
-{
-  "Kestrel": {
-    "Endpoints": {
-      "MyHttpsEndpoint": {
-        "Url": "https://localhost:5001",
-        "ClientCertificateMode": "AllowCertificate",
-        "Certificate": {
-          "Path": "<path to .pfx file>",
-          "Password": "$CREDENTIAL_PLACEHOLDER$"
-        }
-      }
-    }
-  }
-}
-```
+:::code language="csharp" source="~/fundamentals/servers/kestrel/samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_TlsHandshakeCallbackOptions":::
 
-> [!WARNING]
-> In the preceding example, the certificate password is stored in plain-text in `appsettings.json`. The `$CREDENTIAL_PLACEHOLDER$` token is used as a placeholder for the certificate's password. To store certificate passwords securely in development environments, see [Protect secrets in development](xref:security/app-secrets). To store certificate passwords securely in production environments, see [Azure Key Vault configuration provider](xref:security/key-vault-configuration).
+## IIS endpoint configuration
```

Comment:
Why is this in an article about Kestrel configuration?

Reply:
+1. I'd like to understand how it ended up here. If there isn't a good reason then I say dump it.

Reply:
I don't know when it was added.  I'll dump it.